### DISCUSSION TITLE
Rad Pro: Ideas for improvements

### DISCUSSION THREAD

**Discussion by GeigerMuller** [Gissio/radpro#159] [Ideas]
Hi,

I have an idea for improvement. Add the line "device firmware / device fw /device fw version : " to menu>statistic>

Instead of the battery icon on the main screen, you can get numerical information in % about the state of the battery.

**Comment by Gissio** [comment]
The firmware version shows up on device start up.

**Reply by hackra76** [reply]
Well, after updating to 2.0.3, the welcome screen just flashes very briefly, I am not sure if it just my unit, or something went wrong. I tried reinstalling the firmware, but with no effect. My unit is the original GC-01 with APM CPU.

**Reply by Gissio** [reply]
Could you make a short video and post it here?

**Reply by hackra76** [reply]
Here it is

https://github.com/user-attachments/assets/e4c83d0a-e28c-4e58-b4c3-470cc816b27d

...

**Reply by Gissio** [reply]
Thanks for the video. Yes, that is unexpected! Let me check.

**Reply by dc1rdb** [reply]
My GC-01 with APM (Geehy) shows the same behaviour with 2.0.3

**Reply by GeigerMuller** [reply]
Hi, this is the reason why I wrote this. I also have GC-01 with apm32f103rbt6 and JT-RAD01 with CH32F103R8T6. On both, the information with the version only flashes when turned on. For me, it's great that the device boots quickly. I would like it if a line with fw info was added to statistics.

**Comment by ugopapetti** [comment]
Hi, I am observing the same behaviour with original Fosi gc01,  APM
processor, display model GTM024-05 V2.0.

Same was with the previous release  2.0.2, so nothing new...

The flash Is also happening when the display Is placed in low light (
night) mode (in this mode Is very annoying when the illumination Is
timed-out and you press a key to resume back the display...).

Gissio, really compliments for the high quality knowledge and work placed
in this project! Thank you.

Ugo

Il gio 28 nov 2024, 15:51 GeigerMuller ***@***.***> ha
scritto:

**Reply by Gissio** [reply]
Si ringrazia!

**Reply by ugopapetti** [reply]
:-)

**Reply by Gissio** [reply]
Hai detto che c'è anche un flash quando attivi il tuo dispositivo in modalità notte. Potresti fare un video di questo?

**Reply by ugopapetti** [reply]
https://github.com/user-attachments/assets/e183a075-e918-4411-bbdf-081ba06458e7

Here It Is...

**Reply by Gissio** [reply]
Thanks for the video! That's a different problem. Let me check.

**Reply by Gissio** [reply]
If you enable "Pulses" > "Display flashes", does the problem go away?

When the screen goes to sleep, does "Display flashes" work?

**Reply by ugopapetti** [reply]
Hi Gissio,
yes to both questions, no white flashes if "Display flashes" enabled.

While here a minor cosmetical: in "cumulative" screen, where "counts are shown, the counts are using a "comma" number, i.e. the count number one after the reset is shown as "1.000", and so on; should be instead "0001", without any decimal comma, as this quantiti, counts, is an integer (yes, it's a detail).

**Reply by Gissio** [reply]
@GeigerMuller, @dc1rdb, @hackra76, @ugopapetti & others:

can you confirm this works?

[radpro-firmware-snapshot-2.0.4](https://github.com/Gissio/radpro/actions/runs/12149645644/artifacts/2270401201)

Updates are listed here:

https://github.com/Gissio/radpro/blob/main/RELEASES.md

**Reply by dc1rdb** [reply]
Hi Gissio, version 2.0.4 works fine on my GC-01 with APM/Geehy. The splash screen is back and no more flashes when returning from sleep.
Many thanks for your efforts!

**Reply by GeigerMuller** [reply]
Hi, I currently have 2 types gc01(APM32F103R8) and jt-rad01(CH32F103R8). 

The flickering of the display no longer occurs with radpro 2.0.4

A "short" note on the installation/upgrade - on gc01(APM32F103R8) I could not connect to the win 10 pc, it only showed as COM7 network device and putty said ERROR. It also didn't work under Linux. The device jt-rad01 (CH32F103R8) had no problem after setting data mode on. The problem was in the end the cable that works with one device and not with the other. I have original cables for the devices. So in the end I used another cable that I have for the phone. I already had it in my hand tin soldering iron and st-link :D

**Comment by ugopapetti** [comment]
See Just now, let me check.

Thanks a lot for the acceptance of advices.

Il mer 4 dic 2024, 00:30 Gissio ***@***.***> ha scritto:

**Comment by ugopapetti** [comment]
Hi Gissio,
loaded by dragging and dropping V. 2.0.4 with absolutely no issues on my
GC-01 APM using Win 10.
The flashing resuming after display sleep is solved.
The pulse counter (scaler) function now is starting from 0 with no decimal
digits, OK.

When installing a new release, device-specific data are not kept; as an
example, in "statistics", "tube life time" is reset.
Could these basic HW settings be in a non-volatile area of the
microprocessor?
Not all parameters are important, however "tube life time" and "HV profile"
are critical. (of course, if this means that you need to completely rewrite
long parts of the FW, let's forget and survive with the current
situation...).
Gissio, I have further ideas for new requirements, need before to better
define them mathematically and from the usage perspective; given that I
don't want to: overload you or monopolize the arena, tell me if you accept
some further hints or consider the current a "semi-final" release.

Thank You for the excellent firmware, a reference in the cheap geiger arena.

Cheers ugo.

Il giorno mer 4 dic 2024 alle ore 12:48 GeigerMuller <
***@***.***> ha scritto:

> Hi, I currently have 2 types gc01(APM32F103R8) and jt-rad01(CH32F103R8).
>
> The flickering of the display no longer occurs.
>
> A "short" note on the installation/upgrade - on gc01(APM32F103R8) I could
> not connect to the win 10 pc, it only showed as COM7 network device and
> putty said ERROR. It also didn't work under Linux. The device jt-rad01
> (CH32F103R8) had no problem after setting data mode on. The problem was in
> the end the cable that works with one device and not with the other. I have
> original cables for the devices. So in the end I used another cable that I
> have for the phone. I already had it in my hand tin soldering iron and
> st-link :D
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Gissio/radpro/discussions/159#discussioncomment-11460422>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKWI7RBVYT22WUAN2NE6UO32D3TYLAVCNFSM6AAAAABSO3ZC5OVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBWGA2DEMQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Ugo Papetti
***@***.***
Tel. 3357368922

**Reply by Gissio** [reply]
Dimmi (si può essere, in italiano).

**Comment by ugopapetti** [comment]
Gissio,
una paginetta per descrivere.
Grazie del tuo tempo.
ugo

Il giorno mer 4 dic 2024 alle ore 15:25 Gissio ***@***.***>
ha scritto:

> Dimmi (si può essere, in italiano).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Gissio/radpro/discussions/159#discussioncomment-11462120>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKWI7REXC3PBTO3CEVA6MRD2D4GG7AVCNFSM6AAAAABSO3ZC5OVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBWGIYTEMA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Ugo Papetti
***@***.***
Tel. 3357368922

**Comment by ChyrkaA** [comment]
@Gissio  Where can I download firmware version 2.0.4? It's not in the Releases, the latest one is 2.0.3

**Comment by Hantervit** [comment]
Hi,

I have an idea for improvement. 
The Bosean FS-5000's factory firmware includes a compact mode. The screenshot shows what it looks like. In this mode there is only this screen and graph. If the dosimeter is used to detect weak sources of radiation, such as contaminated food, this mode may be useful. It allows you to see all the necessary indicators on one screen at once. Instantaneous, Average, Cumulative and Particle count in cps&cpm.
![Bosean_FS-5000](https://github.com/user-attachments/assets/3c390f12-0fd9-4673-a24d-0cc7b0bb69e8)
